includes/discovery/applications.inc.php

```diff
@@ -98,12 +98,10 @@
     echo str_repeat('-', $num);
     $vars = $apps_to_remove;
     array_unshift($vars, $device['device_id']);
-    dbDelete(
-        'applications',
-        '`device_id`=? AND `app_type` IN ' . dbGenPlaceholders($num),
-        $vars
-    );
     foreach ($apps_to_remove as $app) {
+        $app_obj = Application::withTrashed()->firstOrNew(['device_id' => $device['device_id'], 'app_type' => $app]);
```

Comment:
You don't want withTrashed() in this query.

Reply:
or firstOrNew.

Something like this:
```php
$appToDelete = Application::firstWhere(['device_id' => $device['device_id'], 'app_type' => $app]);
if ($appToDelete) {
   $appToDelete->delete();
}
```

or
```php
DeviceCache::getPrimary()->applicatins()->whereIn('app_type', $apps_to_remove)->get()->each->delete();
```

We could call `DeviceCache::getPrimary()->applicatins()->whereIn('app_type', $apps_to_remove)->delete();`, but that wouldn't fire events for the device observer.

Reply:
Yeah, went and changed this to...

```
        Application::where([
            ['device_id', $device['device_id']],
            ['app_type', $app],
        ])->delete()->save();
```